### PR TITLE
Merged release notes & docs

### PR DESCRIPTION
**Pull request by montegoulding**

### CODE REVIEW

**Comment by montegoulding** [comment]
This patch needs rebasing on #5895

File: builder/builder_utilities.livecodescript [hunk 1]
```diff
@@ -637,20 +638,210 @@ end builderTSNetUnpack
...
+An commma delimited list
+
+**/
+function builderComponentCategories
+   return "engine,IDE,extension"
+end builderComponentCategories
+
+/**
+
+Return the names of editions in notes priority order
```
Comment (montegoulding) [inline builder/builder_utilities.livecodescript]: This can be deleted

File: builder/release_notes_builder.livecodescript [hunk 2]
```diff
@@ -30,24 +34,164 @@ command releaseNotesBuilderRun pEdition, pVersion, pReleaseType, pOutputDir
...
+               next repeat
+            end if
+            
+            repeat for each line tFolder in FileGetSubFolders(tTypePath)
+               put the number of elements of tComponents + 1 into tIndex
+               put tFolder into tComponents[tIndex]["folder"]
+               put "extension" into tComponents[tIndex]["metadata"]["category"]
+               
+               if tPath is  builderPrivateRepoFolder() then
+                  put "community extra" into tComponents[tIndex]["metadata"]["edition"]
```
Comment (livecodeali) [inline builder/release_notes_builder.livecodescript]: This may just be an internal thing, but it still uses "community extra"

**Comment by livecodeali** [comment]
@montegoulding Might be worth an interactive rebase to fixup `[[ CommunityPlus ]] Add DMG backgrounds` into `[[ CommunityPlus ]] [[ TESTING ]] Add tmp dmg backgrounds` and change the commit name

**Comment by livecodeali** [comment]
@livecode-vulcan review ok c8a4fb3

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok c8a4fb32c56f8c672fae10c58bb39a303b7ac27a

**Comment by livecode-vulcan** [comment]
😎 test success c8a4fb32c56f8c672fae10c58bb39a303b7ac27a

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/739)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/755)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/729)
* try-community-universal-ios-iphoneos10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.3/builds/672)
* try-community-universal-ios-iphonesimulator10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.3/builds/674)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/733)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/753)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/738)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/755)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/737)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/748)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/769)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/733)

**Pull request merged**